pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -365,6 +376,31 @@ def post_training_step(self):
         if not self.lightning_module.automatic_optimization:
             self.model.require_backward_grad_sync = True
 
+    def add_to_queue(self, trainer: "pl.Trainer", queue: torch.multiprocessing.SimpleQueue) -> None:
+        """
+        Appends the :attr:`trainer.callback_metrics` dictionary to the given queue.
+        To avoid issues with memory sharing, we cast the data to numpy.
+
+        Args:
+            queue: the instance of the queue to append the data.
+        """
+        callback_metrics: dict = apply_to_collection(
```

Comment:
Can we add the ModelCheckpoint state too ? So best_score is available, and this should properly enable Optuna use-case ?

Reply:
yes, will add in this PR!

Reply:
actually on second thought I think it makes more sense to have separate PR

Reply:
@tchaton for this how exactly can I access the state from the callback? i.e. I have `trainer.checkpoint_callback`. I can do `trainer.checkpoint_callback.best_model_score`, but I don't see any property that holds a multi-field state.

Reply:
I was thinking about something like this.

```py
def add_to_queue(self, queue):
       states = [cb.on_save_checkpoint() for cb in trainer.callbacks if isinstance(cb, ModelCheckpoint)]
       queue.put(states)

def ...
```